src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemResources.cs

```diff
@@ -2028,6 +2029,122 @@ internal override bool IsUnset
         #endregion Properties
     }
 
+    internal class DeferredResourceReferenceList : IEnumerable
```

Comment:
I'm not sure what guidelines this repo follows, but it's generally preferred to have one type per file. This helps with general readability (a developer doesn't have to scroll two thousand lines), repo maintenance and discoverability (to find type ABC one needs to search for ABC file).
An added bonus, you can apply nullable annotations at the top of the file (as opposed to using #nullable enable #nullable disable intermingled with the code).

Reply:
Sorry for my late response.
While i think you are right, i just followed the current state of the file.
It already contains 4 classes which seem to be related to each other.
Splitting my class out to a separate file didn't feel right here.

Reply:
Made it sealded.